src/RamStatement.h

```diff
@@ -155,10 +155,21 @@ class RamStore : public RamRelationStatement {
         }
         const auto& outputDirectives = getRelation().getOutputDirectives();
         os << "STORE DATA FOR " << getRelation().getName() << " TO {";
-        if (!outputDirectives.empty()) os << outputDirectives.begin()->getFileName();
-        if (outputDirectives.size() > 1)
-            std::for_each(outputDirectives.begin() + 1, outputDirectives.end(),
-                    [&os](IODirectives directives) { os << ", " + directives.getFileName(); });
+        bool useStdout = false;
+        std::for_each(
+                outputDirectives.begin(), outputDirectives.end(), [&useStdout](IODirectives directives) {
+                    if (directives.get("IO") == "stdout") useStdout = true;
```

Comment:
This seems like it will still fail for any IO that is not to stdout or file. I think it would be better to either output `directives.get("IO")` with an extra output if using file IO, or output all the directives.

e.g. `STORE DATA FOR relname TO {IO=stdout,delimiter='xx'}`
or for more complicated cases,
`STORE DATA FOR relname TO {{IO=stdout,delimiter='xx'},{IO=file,delimiter='xx',filename='bob.log}}`

Reply:
Good plan, I'll modify it to write all of the IO directives (for inputs as well as outputs of course).